### PR TITLE
Horde Invasion

### PR DESCRIPTION
**Pull request by Malicos**
## Changelog:
- **The Horde Invasion:**
   - The Horde is scripted to conquer Azeroth and Khaz Modan. Its Warchief and vassals get special troops. The Warchief gets the Great Invader modifier giving high dread.
   - The Horde CB gives 50% of conquered duchies to the strongest clans. The rest goes to the Warchief.
   - The Horde falls apart if it takes enough lands for the orcs or the year is 613.
- **Title name changes:**
   - Orcish feudal and clan counts are called **_Overlords_**.
   - Orcish feudal and clan dukes are called **_High Overlords_**.
   - Rulers of the Horde are called **_Warchiefs_**.
   - Orcish kings are called **_Warlords_**.
   - The Horde is no longer **_High Kingdom of Horde_**.
- The orcish clans and the Horde start with the **_Orcish Elective_** law: a marshal, knights and powerful vassals can vote and can be elected.
- The Warchief can use the **_Invite Remaining Clans_** decision summoning the Burning Blade and Dragonmaw clans and getting more special troops.

## Developer changelog:
- Added `purge_historical_character_effect`.

# How to test:
Observe or play the Horde in the 583 start date and tell your opinion.
[Click Here](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/20/files?file-filters%5B%5D=.yml) to proofread localization files.

### CODE REVIEW

**Review by Deuce20201 — COMMENTED** [review]
My issue is that the Horde just steamrolls everything quite easily. I've observed 5 times now and the Horde just takes Stormwind, Shadowforge and Ionforge at the latest 593; and then breaks apart like you said it would. 

Lore friendly I suppose, but I think it might be an issue if you're playing as Stormwind, let's say, in the 583 start date

**Comment by zumbak04** [comment]
> My issue is that the Horde just steamrolls everything quite easily. I've observed 5 times now and the Horde just takes Stormwind, Shadowforge and Ironforge at the latest 593; and then breaks apart like you said it would.
> 
> Lore friendly I suppose, but I think it might be an issue if you're playing as Stormwind, let's say, in the 583 start date

I'll be balancing it when the Alliance is done.

**Review by Deuce20201 — COMMENTED** [review]
Also, why does the Horde just fall apart after Ironforge? They continue North after that in lore. Is that a balancing decision?

**Comment by zumbak04** [comment]
> Also, why does the Horde just fall apart after Ironforge? They continue North after that in lore. Is that a balancing decision?

North in Grim Batol?

**Comment by Deuce20201** [comment]
> > Also, why does the Horde just fall apart after Ironforge? They continue North after that in lore. Is that a balancing decision?
> 
> North to Grim Batol?

Well they go as North as Capital City

**Comment by zumbak04** [comment]
> Well they go as North as Capital City

They fought with Lordaeron, that's why. The orcs haven't settled further than Khaz Modan.

**Comment by Malicos** [comment]
> 
> 
> > Also, why does the Horde just fall apart after Ironforge? They continue North after that in lore. Is that a balancing decision?
> 
> North to Grim Batol?

They actually should invade anyone holding Hillsbrad after that.

**Comment by Deuce20201** [comment]
> > > Also, why does the Horde just fall apart after Ironforge? They continue North after that in lore. Is that a balancing decision?
> > 
> > 
> > North to Grim Batol?
> 
> They actually should invade anyone holding Hillsbrad after that.

Lordaeron holds Hillsbrad at the time I think. So I don't think the Horde should just fall apart after Ironforge. Should be scripted that they go to war with Lordaeron as well.

**Comment by zumbak04** [comment]
> Lordaeron holds Hillsbrad at the time I think. So I don't think the Horde should just fall apart after Ironforge. Should be scripted that they go to war with Lordaeron as well.

It will make the Horde blob too big, destroying all the human kingdoms.

**Comment by Deuce20201** [comment]
> > Lordaeron holds Hillsbrad at the time I think. So I don't think the Horde should just fall apart after Ironforge. Should be scripted that they go to war with Lordaeron as well.
> 
> It will make the Horde blob too big, destroying all the human kingdoms.

True. However you said you'd balance it all when you create the Alliance.

**Comment by zumbak04** [comment]
> True. However you said you'd balance it all when you create the Alliance.

Balance the Horde troops.

**Review by TheGroxEmpire — COMMENTED** [review]
- The Horde are releasing hostages after enforcing demand, which is pretty not in the Old Horde spirit (Zug zug). They should keep any hostages from the war.
<details>

image
</details>

- Should make it like a Holy War CB where ruler of opposing religion (or even culture) within de-jure will be deposed. This way we could avoid having human vassal as the Horde.
<details>


image
</details>

- More often than not, the Horde were having civil war immediately after winning the war. And sometimes the AI granted them independence. I think the likeliness for a revolt (At least peasant / populism revolt) should be hugely decreased after invasion for a bit.
<details>


image
</details>

- You could invade **any** bordering kingdom. Should probably limit it to Azeroth and Khaz Modan.
<details>

image
image
</details>

**Comment by zumbak04** [comment]
> * The Horde are releasing hostages after enforcing demand, which is pretty not in the Old Horde spirit (Zug zug). They should keep any hostages from the war.

There's a gameplay reason why they're released. You can't declare war if the war target is imprisoned and you get a score for having prisoners.

> * Should make it like a Holy War CB where ruler of opposing religion (or even culture) within de-jure will be deposed. This way we could avoid having human vassal as the Horde.

The Horde usurps all de-jure lands. These vassals are the result of the Horde usurping (and later destroying Stormwind title). Usually, there's no many of them, so it won't hurt the Horde much.

> * You could invade **any** bordering kingdom. Should probably limit it to Azeroth and Khaz Modan.

I would keep it up to the player. The AI will stick to Azeroth and Khaz Modan.

**Comment by zumbak04** [comment]
> * More often than not, the Horde were having civil war immediately after winning the war. And sometimes the AI granted them independence. I think the likeliness for a revolt (At least peasant / populism revolt) should be hugely decreased after invasion for a bit.

I buffed the Warchief's dread and gave more special troops. The Horde should feel more save now.

**Review by TheGroxEmpire — COMMENTED** [review]
> There's a gameplay reason why they're released. You can't declare war if the war target is imprisoned and you get a score for having prisoners.

I didn't think of that, makes sense.

> The Horde usurps all de-jure lands. These vassals are the result of the Horde usurping (and later destroying Stormwind title). Usually, there's no many of them, so it won't hurt the Horde much.

You're right there's not many of them. But would it be a big hassle to include the similar usurpation Paradox used in holy war CB?
<details>

		#go through the dejure hierarchy under target titles, transfer titles with same or worse tolerance holders (their religion equaly or less tolerated than the defender's),
		# take the holder as vassal otherwise and don't go deeper

		every_in_list = {
			list = target_titles
			custom_tooltip = RELIGIOUS_CB_TITLE

			conquest_cb_title_transfer = {
				RELIGIOUS_WAR = yes
			}
		}

		every_in_list = {
			list = vassals_taken
			change_liege = {
				liege = scope:attacker
				change = scope:change
			}
		}
		
		every_in_list = {
			list = titles_taken
			change_title_holder = {
				holder = scope:attacker
				change = scope:change
				take_baronies = yes
			}
		}

		resolve_title_and_vassal_change = scope:change

</details>

> I would keep it up to the player. The AI will stick to Azeroth and Khaz Modan.

Understandable, the AI seems to stick on the predetermined path.

> I buffed the Warchief's dread and gave more special troops. The Horde should feel more save now.

This seems to be working. Horde is now consistently staying in-tact until breakup now.

- a21755bd9d5922a7eeee3599ac594d46e8a6718b changed Orcish kings title to "Warlord". I have suggestion for feudal title:
King: High Warlord / Warlord
Duke: Warlord / High Overlord
Count: High Overlord / Overlord

**Comment by zumbak04** [comment]
> You're right there's not many of them. But would it be a big hassle to include the similar usurpation Paradox used in holy war CB?

The holy war doesn't usurp the targeted title.
Like here, I targeted France.
<details>
<summary>Click to expand</summary>

image

</details>

And we need the targeted title usurpation to put these peripheral vassals under the Horde so it's less bordergore.

**Comment by zumbak04** [comment]
> * [a21755b](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/commit/a21755bd9d5922a7eeee3599ac594d46e8a6718b) changed Orcish kings title to "Warlord". I have suggestion for feudal title:
>   King: High Warlord / Warlord
>   Duke: Warlord / High Overlord
>   Count: High Overlord / Overlord

Done in 708616ba947e46556b2a6a9db5b322350bb9e06f.

**Review by TheGroxEmpire — APPROVED** [review]
All things considered then.

**Review by Deuce20201 — APPROVED** [review]
Sorry for my absence yesterday, got busy all of the sudden. 

I still think the Horde should go to war with Lordearon (or at least Stromgarde) instead of breaking apart at Ironforge. 

But the changes so far are still very good.

File: localization/english/dynasties/wc_dynasty_names_l_english.yml [hunk 1]
```diff
@@ -6918,4 +6916,4 @@
  dynn_Kosumothid:0 "Kosumothid"
  dynn_Yenajzid:0 "Yenajzid"
  dynn_Erudaxid:0 "Erudaxid"
- dynn_Zakajzid:0 "Zakajzid"
\ No newline at end of file
+ dynn_Zakajzid:0 "Zakajzid"l_english:
```
Comment (arithon) [inline localization/english/dynasties/wc_dynasty_names_l_english.yml:6919]: is this intended?

**Review by arithon — COMMENTED** [review]
almost perfect :)

File: localization/english/dynasties/wc_dynasty_names_l_english.yml (hunk 1 above)
Comment (zumbak04) [reply localization/english/dynasties/wc_dynasty_names_l_english.yml]: Fixed in b279cfe17d70a9ca4d9bd3382b71f371cdac6613.

File: localization/english/wc_horde_invasion_l_english.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿l_english:
+ WCHOI_2_TITLE:0 "The Invasion from Draenor"
+ WCHOI_2_DESC:0 "A storm brews over the southern reaches of Azeroth, within the Black Morass. A gateway, the Dark Portal, has been opened to a dying world known as Draenor — home to wild beasts, orcs, ogres and many other vile creatures. Bloodthirsty and in search of new lands, the orcish clans have united beneath a powerful Warchief who rules the Horde with an iron first. Now they seek to go through the Dark Portal and conquer Azeroth!"
+ WCHOI_2_OPT_A:0 "Lok'tar ogar!"
+ WCHOI_2_OPT_B:0 "Let us wait and see what happens."
+ WCHOI_2_OPT_C:0 "Something must be done about this!"
+ 
+ WCHOI_3_DESC:0 "It's time to stop this carnage. The Horde has conquered enough land to start a new life in Azeroth. Many orcs are returning to their families, burying their axes into the ground. They have vowed to forget the horrors of war that they have had to go through. The warlords of the Horde still pose a threat to some parts of Azeroth, but the bloodiest wars are behind us since the clans no longer see the point in keeping the united Horde."
```
[the remark this answers is not in the document]
Comment (zumbak04) [reply localization/english/wc_horde_invasion_l_english.yml]: Done in 8430f21b0f45ad0238a2498fa5adf848fada1dd6.

**Review by arithon — APPROVED** [review]
loc LGTM

**Pull request merged**